### PR TITLE
Checkbox TS Convert

### PR DESCRIPTION
**Pull request by jaketrent**
## Instructions

- fill in title: format of "<package name>: short description"
- label: attach bug or enhancement request label

### What You're Solving

- Any links to #57 related issues
- Explanation of the purpose of the code

### Design Decisions

- Why you made certain design decisions
- Anything to watch out for
- Any input you're seeking related to the design
- Tradeoffs you conciously made

### How to Verify

- Steps needed to independently verify additions

### CODE REVIEW

File: packages/checkbox/src/react/index.tsx [hunk 1]
```diff
@@ -89,60 +122,43 @@ const Checkbox = React.forwardRef((props, forwardedRef) => {
...
+        tabIndex={-1}
         type="checkbox"
         value={props.value}
-        {...styles.input(themeName, props)}
-        {...filterReactProps(props, { tagName: 'input' })}
+        {...styles.input()}
+        {...omit<CheckboxProps, ['label', 'onCheck', 'error', 'indeterminate']>(
+          props,
+          ['label', 'onCheck', 'error', 'indeterminate']
+        )}
```
Comment (danethurber) [inline packages/checkbox/src/react/index.tsx:132]: 👍 

there is probably a way to make the `omit` function figure these generic args out without the added code but i'm not that good at ts

File: packages/checkbox/tsconfig.build.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    ".storybook",
+    "node_modules",
+    "dist",
+    "**/*.spec.*",
+    "**/*.story.*"
+  ],
+  "compilerOptions": {
+    "declaration": true,
+    "outDir": "dist",
+    "rootDir": "src"
```
Comment (danethurber) [inline packages/checkbox/tsconfig.build.json:14]: `strict`?

File: packages/checkbox/src/react/__specs__/__snapshots__/storyshots.spec.ts.snap [hunk 3]
```diff
@@ -172,19 +167,17 @@ exports[`Storyshots Checkbox disabled & errored 1`] = `
...
       <div
-        aria-checked={false}
         data-css-1lxtjk6=""
         role="checkbox"
-        tabIndex="-1"
+        tabIndex={-1}
       />
     </div>
     <input
-      checked={false}
```
Comment (danethurber) [inline packages/checkbox/src/react/__specs__/__snapshots__/storyshots.spec.ts.snap:182]: why remove?

File: packages/checkbox/src/react/__specs__/__snapshots__/storyshots.spec.ts.snap [hunk 4]
```diff
@@ -172,19 +167,17 @@ exports[`Storyshots Checkbox disabled & errored 1`] = `
       data-css-1xpuf1q=""
     >
       <div
-        aria-checked={false}
```
Comment (danethurber) [inline packages/checkbox/src/react/__specs__/__snapshots__/storyshots.spec.ts.snap:175]: why remove?

File: packages/checkbox/src/react/__specs__/__snapshots__/storyshots.spec.ts.snap [hunk 5]
```diff
@@ -279,10 +270,9 @@ exports[`Storyshots Checkbox error 1`] = `
       <input
         checked={true}
         data-css-gyrv08=""
-        disabled={false}
```
Comment (danethurber) [inline packages/checkbox/src/react/__specs__/__snapshots__/storyshots.spec.ts.snap:282]: why remove?

File: packages/checkbox/src/react/index.tsx [hunk 6]
```diff
@@ -1,42 +1,75 @@
...
+  extends React.HTMLAttributes<HTMLDivElement>,
+    Record<string, unknown> {
+  checked?: boolean
+  disabled?: boolean
+  error?: boolean
+  indeterminate?: boolean
+  label: React.ReactNode
+  name?: string
+  onCheck?: (
+    evt: Event,
```
Comment (danethurber) [inline packages/checkbox/src/react/index.tsx:49]:
```suggestion
    evt: React.ChangeEvent<HTMLInputElement>,
```

maybe?

File: packages/checkbox/src/react/index.tsx [hunk 7]
```diff
@@ -1,42 +1,75 @@
...
+    checked: boolean,
+    value: string | number,
+    name: string | undefined
```
Comment (danethurber) [inline packages/checkbox/src/react/index.tsx:52]:
```suggestion
    name?: string
```

maybe?

File: packages/checkbox/src/react/index.tsx [hunk 8]
```diff
@@ -1,42 +1,75 @@
...
+  checked?: boolean
+  disabled?: boolean
+  error?: boolean
+  indeterminate?: boolean
+  label: React.ReactNode
+  name?: string
+  onCheck?: (
+    evt: Event,
+    checked: boolean,
+    value: string | number,
```
Comment (danethurber) [inline packages/checkbox/src/react/index.tsx:51]:
```suggestion
    value: React.ReactText,
```

we've done this with other form elements?

File: packages/checkbox/src/react/index.tsx [hunk 9]
```diff
@@ -89,60 +122,43 @@ const Checkbox = React.forwardRef((props, forwardedRef) => {
...
-  onCheck: PropTypes.func,
-  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired
-}
-
-Checkbox.defaultProps = {
-  checked: false,
-  disabled: false,
-  error: false,
-  indeterminate: false
-}
```
Comment (danethurber) [inline packages/checkbox/src/react/index.tsx:119]: are the default props being assigned somewhere. is there why the snapshots changed. do we even need these anymore?

**Review by danethurber — CHANGES_REQUESTED** [review]

File: packages/checkbox/tsconfig.build.json (hunk 2 above)
Comment (jaketrent) [reply packages/checkbox/tsconfig.build.json:14]: ah yes, thank you.  I forgot!

File: packages/checkbox/src/react/__specs__/__snapshots__/storyshots.spec.ts.snap (hunk 5 above)
Comment (jaketrent) [reply packages/checkbox/src/react/__specs__/__snapshots__/storyshots.spec.ts.snap:282]: This is apparently just a difference with how the `...filterReactProps` and `...props` behavior works.

disabled=false === enabled, so it's not really required anyway

File: packages/checkbox/src/react/index.tsx (hunk 9 above)
Comment (jaketrent) [reply packages/checkbox/src/react/index.tsx:119]: My thought that defaults that are false for boolean types are exactly needed anyway.  Undefined is falsy too.

File: packages/checkbox/src/react/index.tsx (hunk 7 above)
Comment (jaketrent) [reply packages/checkbox/src/react/index.tsx:52]: I'd rather keep it as written for the reason that:

`paramName?` means that the function might be called without that arg.

In this case, onCheck is always called with a 4th arg:

https://github.com/pluralsight/design-system/blob/cabaffb86bba25789ef63777ca07c42c7e603203/packages/checkbox/src/react/index.tsx#L92

But that arg might be undefined.

File: packages/checkbox/src/react/index.tsx (hunk 1 above)
Comment (jaketrent) [reply packages/checkbox/src/react/index.tsx:132]: That would be awesome.  This was repetitive and didn't feel great.

**Review by danethurber — APPROVED** [review]
👍

**Comment by jaketrent** [comment]
K, submitted again. Ready for another review. Did the strict mode stuff.  Used badge a simpler case to figure out some of the strict mode stuff, so it's included as well.

**Pull request merged**